lns_hfs.py

```diff
@@ -168,9 +168,79 @@ def main(args):
     scaled_objective = scale*(best_objective+offset)
     scaled_lower_bound = scale*(lower_bound+offset)
 
+    verify_solution(tmp_hfs_file, tmp_sol_file, best_objective)
+
+    remove_tmp_file(tmp_hfs_file)
+    remove_tmp_file(tmp_sol_file)
+
     print('BQP_DATA, %d, %d, %f, %f, %f, %f, %f, %d, %d' % (nodes, edges, scaled_objective, scaled_lower_bound, best_objective, lower_bound, best_runtime, 0, best_nodes))
 
 
+def create_tmp_file(prefix=None):
+    fd, filename = tempfile.mkstemp(prefix=prefix, dir=HFS_DIR)
+    os.close(fd)
+    return filename
+
+
+def remove_tmp_file(filename):
+    print('INFO: removing file {}'.format(filename), file=sys.stderr)
+    try:
+        os.remove(filename)
+    except:
+        print('WARNING: removing file {} failed'.format(filename), file=sys.stderr)
+
+
+def verify_solution(tmp_hfs_file, tmp_sol_file, best_objective):
```

Comment:
This is a good validation to make, but what if there is an inconsistency between the bqpjson file and the HFS data model (e.g. a difference in the objective function's energy due to finite precision?).  

In addition to this check I recommend that we take the HFS solution and then evaluate its value using the bqpjson data model.  Then the results in `BQP_DATA` should be consistent with the other solvers that use floating point arithmetic.